### PR TITLE
Implement the type of event dropdown editor for single event edits

### PR DESCRIPTION
**Pull request by Coenego**
screen shot 2015-02-02 at 10 44 03

### CODE REVIEW

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 1]
```diff
@@ -260,6 +260,39 @@ define(['gh.api.event', 'gh.api.groups', 'gh.api.series', 'gh.api.util', 'gh.adm
             // Show the save button
             toggleSubmit();
         }
+        // Return the selected value
+        return value;
+    };
+
+    /**
+     * Verifies that a valid value for event type was entered and persists the value in the field
```
Comment (bp323) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: `Verify`

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 2]
```diff
@@ -260,6 +260,39 @@ define(['gh.api.event', 'gh.api.groups', 'gh.api.series', 'gh.api.util', 'gh.adm
...
+     *
+     * @param  {String}   value     The new value for the item
+     * @return {String}             The value to show in the editable field after editing completed
+     * @private
+     */
+    var editableEventTypeSubmitted = function(value, editableField) {
+        // Get the value
+        value = $.trim(value);
+        // Get the event ID. If not eventId is found on the tr that means we're dealing with a newly added event
```
Comment (bp323) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: `If no eventId`

File: shared/gh/js/controller/gh.admin-event-type-select.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        'content': function(string, settings, original) {
+            var eventType = $(original).attr('data-type');
+            if (eventType) {
+                $(this).find('select option[value="' + eventType + '"]').attr('selected', 'selected');
+            }
+        },
+        'plugin': function(settings, original) {
+            $('select', this).on('change', function() {
+                $(this).parent().find('input[type="hidden"]').val($(this).val());
```
Comment (bp323) [inline shared/gh/js/controller/gh.admin-event-type-select.js]: Can you add some documentation to this?

File: shared/gh/js/controller/gh.admin-event-type-select.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                $(this).closest('td').attr('data-type', $(this).val());
+                $(this).closest('td').attr('data-first', $(this).val().substr(0,1));
+                $(this).closest('form').trigger('submit');
+            });
+            $('select', this).on('focusout', function() {
+                $(original).removeClass('gh-editing');
```
Comment (bp323) [inline shared/gh/js/controller/gh.admin-event-type-select.js]: Some docs here as well

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 5]
```diff
@@ -295,6 +329,15 @@ define(['gh.api.event', 'gh.api.groups', 'gh.api.series', 'gh.api.util', 'gh.adm
...
             }
         });
+
+        // Apply jEditable to the event notes
+        $('.gh-jeditable-events-select').editable(editableEventTypeSubmitted, {
+            'cssclass' : 'gh-jeditable-form',
+            'placeholder': '',
+            'select': true,
+            'tooltip': 'Click to edit event notes',
+            'type': 'event-type-select'
```
Comment (bp323) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: Can you add a callback here that focuses the original element when the form has been submitted? Currently focus is lost.

File: apps/timetable/admin/ui/css/skin.css [hunk 6]
```diff
@@ -209,6 +209,36 @@
...
+.gh-batch-edit-events-container .gh-event-type[data-type="Lecture"]::before {
+    background-color: #19A3FF;
+}
+
+.gh-batch-edit-events-container .gh-event-type[data-type="Paper"]::before {
+    background-color: #62326E;
+}
+
+.gh-batch-edit-events-container .gh-event-type[data-type="Seminar"]::before {
+    background-color: #92308A;
```
Comment (bp323) [inline apps/timetable/admin/ui/css/skin.css]: Paper and Seminar have very similar colours, maybe we can make one of them red-ish?

**Comment by bp323** [comment]
Looks and works fantastic! Only a few minor comments remaining, back to @Coenego

**Comment by Coenego** [comment]
Reassigning to @bp323 for review after follow-up

**Comment by bp323** [comment]
Great work, looks awesome. Verified and merged.

**Pull request merged**